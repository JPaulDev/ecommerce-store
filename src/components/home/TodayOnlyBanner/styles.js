import styled, { css } from 'styled-components';

const headingStyles = css`
  font-family: 'Uni Sans', sans-serif;
  font-weight: 600;
`;

export const Container = styled.div`
  background-color: #d72140;
  background-image: url(${({ background }) => background});
  background-position: 96% 60%;
  background-repeat: no-repeat;
  border: 1px solid var(--color-grey-9);
  margin: 15px 0;
  overflow: hidden;
  padding: 20px 220px 27px 20px;
  position: relative;

  @media screen and (min-width: 935px) {
    display: flex;
    align-items: center;
    column-gap: 30px;
    background-image: none;
    background-color: white;
    padding: 25px 30px;
  }
`;

export const RedOval = styled.div`
  background: linear-gradient(to right, #c12134, #da2141);
  border-radius: 50%;
  height: 170px;
  left: -67px;
  position: absolute;
  width: 300px;

  @media screen and (max-width: 934px) {
    display: none;
  }
`;

export const LeftHeading = styled.h3`
  color: white;
  font-size: 1.6rem;
  z-index: 10;
  ${headingStyles}

  @media screen and (min-width: 575px) {
    font-size: 1.95rem;
    margin-right: 30px;
  }
`;

export const RightHeading = styled.h3`
  font-size: clamp(20px, 3.48vw, 25px);
  margin-right: auto;
  ${headingStyles}

  @media screen and (min-width: 700px) {
    font-size: 1.6rem;
  }

  @media screen and (max-width: 934px) {
    margin: 5px 0 15px;
  }
`;
