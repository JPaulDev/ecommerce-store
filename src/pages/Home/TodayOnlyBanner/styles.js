import styled from 'styled-components';

export const Container = styled.div`
  background-color: #d72140;
  background-image: url(${({ background }) => background});
  background-position: 96% 60%;
  background-repeat: no-repeat;
  border: 1px solid var(--color-grey-9);
  margin: 15px 0;
  overflow: hidden;
  padding: 20px 220px 27px 20px;
  position: relative;

  @media screen and (min-width: 935px) {
    display: flex;
    align-items: center;
    column-gap: 30px;
    background-image: none;
    background-color: white;
    padding: 25px 30px;
  }
`;

export const RedOval = styled.div`
  background: linear-gradient(to right, #c12134, #da2141);
  border-radius: 50%;
  height: 170px;
  left: -67px;
  position: absolute;
  width: 300px;

  @media screen and (max-width: 934px) {
    display: none;
  }
`;

export const LeftHeader = styled.h3`
  color: white;
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.8rem;
  font-weight: 600;
  line-height: 1;
  z-index: 10;

  @media screen and (min-width: 575px) {
    font-size: 1.95rem;
    margin-right: 30px;
  }
`;

export const RightHeader = styled.h3`
  font-family: 'Uni Sans', sans-serif;
  font-size: clamp(20px, 3.48vw, 25px);
  font-weight: 600;
  margin-right: auto;

  @media screen and (min-width: 700px) {
    line-height: 1;
    font-size: 1.6rem;
  }

  @media screen and (max-width: 934px) {
    margin: 10px 0 15px;
  }
`;
