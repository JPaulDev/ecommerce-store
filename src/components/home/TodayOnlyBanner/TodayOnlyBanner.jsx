import background from '../../../../public/images/today-only-banner/gaming-pc.webp';
import { LinkChevron, LinkButton, ResponsiveBox } from '../../ui';
import * as Styled from './styles';

export default function TodayOnlyBanner() {
  const linkText = 'View All Offers';

  return (
    <Styled.Container background={background.src}>
      <Styled.RedOval />
      <Styled.Heading>
        TODAY ONLY
        <Styled.InnerText>
          Fantastic offers, updated every weekday
        </Styled.InnerText>
      </Styled.Heading>
      <ResponsiveBox sx={{ display: { 0: 'none', 935: 'block' } }}>
        <LinkChevron href="/" fontSize={13} iconSize={18}>
          {linkText}
        </LinkChevron>
      </ResponsiveBox>
      <ResponsiveBox sx={{ display: { 0: 'block', 935: 'none' } }}>
        <LinkButton href="/">{linkText}</LinkButton>
      </ResponsiveBox>
    </Styled.Container>
  );
}
