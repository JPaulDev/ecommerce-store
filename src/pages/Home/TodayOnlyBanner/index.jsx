import useMediaQuery from '../../../hooks/useMediaQuery';
import backgroundImage from '../../../assets/images/today-only-banner/gaming-pc.webp';
import LinkWithChevron from '../../../components/LinkWithChevron';
import LinkButtonGrey from '../../../components/LinkButtonGrey';
import * as Styled from './styles';

export default function TodayOnlyBanner() {
  const isMatch = useMediaQuery('(min-width: 935px)');
  const linkText = 'View All Offers';

  return (
    <Styled.Container background={backgroundImage}>
      <Styled.RedOval />
      <Styled.LeftHeader>TODAY ONLY</Styled.LeftHeader>
      <Styled.RightHeader>
        Fantastic offers, updated every weekday
      </Styled.RightHeader>
      {isMatch ? (
        <LinkWithChevron url="" text={linkText} fontSize="0.8rem" />
      ) : (
        <LinkButtonGrey href="" padding="12px 25px">
          {linkText}
        </LinkButtonGrey>
      )}
    </Styled.Container>
  );
}
