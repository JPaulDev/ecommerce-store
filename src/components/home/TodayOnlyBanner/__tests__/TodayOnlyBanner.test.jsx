import renderer from 'react-test-renderer';
import TodayOnlyBanner from '../index';

describe('TodayOnlyBanner component', () => {
  it('renders a today only offers banner', () => {
    const tree = renderer.create(<TodayOnlyBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
