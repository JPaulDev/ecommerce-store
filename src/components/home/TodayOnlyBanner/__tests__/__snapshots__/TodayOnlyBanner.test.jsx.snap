// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodayOnlyBanner component renders a today only offers banner 1`] = `
.c7 {
  background: linear-gradient(to top,var(--color-grey-5),var(--color-grey-8));
  border-radius: 6px;
  box-shadow: var(--box-shadow-md);
  color: white;
  display: inline-block;
  font-size: var(--font-size-13);
  font-weight: var(--font-weight-bold);
  padding: 12px 25px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
}

.c7:hover {
  background: linear-gradient( to top, var(--color-grey-6), var(--color-grey-5) );
  box-shadow: var(--box-shadow-hover);
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--color-grey-4);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: var(--font-size-13);
  font-weight: var(--font-weight-bold);
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
}

.c5 svg {
  margin-left: 10px;
}

.c0 {
  background-color: var(--color-red-3);
  background-image: url(/img.jpg);
  background-position: 96% 60%;
  background-repeat: no-repeat;
  border: 1px solid var(--color-grey-12);
  margin: 15px 0;
  overflow: hidden;
  padding: 20px 220px 27px 20px;
  position: relative;
}

.c1 {
  background: linear-gradient(to right,var(--color-red-4),var(--color-red-3));
  border-radius: 50%;
  height: 170px;
  left: -67px;
  position: absolute;
  width: 300px;
}

.c2 {
  color: white;
  font-size: var(--font-size-26);
  z-index: 10;
  font-family: 'Uni Sans',sans-serif;
  font-weight: var(--font-weight-semibold);
}

.c3 {
  font-size: clamp(20px,3.48vw,25px);
  margin-right: auto;
  font-family: 'Uni Sans',sans-serif;
  font-weight: var(--font-weight-semibold);
}

@media screen and (min-width:0px) {
  .c4 {
    display: none;
  }
}

@media screen and (min-width:935px) {
  .c4 {
    display: block;
  }
}

@media screen and (min-width:0px) {
  .c6 {
    display: block;
  }
}

@media screen and (min-width:935px) {
  .c6 {
    display: none;
  }
}

@media screen and (min-width:935px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-column-gap: 30px;
    column-gap: 30px;
    background-image: none;
    background-color: white;
    padding: 25px 30px;
  }
}

@media screen and (max-width:934px) {
  .c1 {
    display: none;
  }
}

@media screen and (min-width:575px) {
  .c2 {
    font-size: var(--font-size-30);
    margin-right: 35px;
  }
}

@media screen and (min-width:700px) {
  .c3 {
    font-size: var(--font-size-26);
  }
}

@media screen and (max-width:934px) {
  .c3 {
    margin: 5px 0 15px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <h3
    className="c2"
  >
    TODAY ONLY
  </h3>
  <h3
    className="c3"
  >
    Fantastic offers, updated every weekday
  </h3>
  <div
    className="c4"
  >
    <a
      className="c5"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      View All Offers
      <svg
        fill="currentColor"
        height={18}
        viewBox="0 0 512 512"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256C0 397.4 114.6 512 256 512s256-114.6 256-256S397.4 0 256 0S0 114.6 0 256zM241 377c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l87-87-87-87c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L345 239c9.4 9.4 9.4 24.6 0 33.9L241 377z"
        />
      </svg>
    </a>
  </div>
  <div
    className="c6"
  >
    <a
      className="c7"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      View All Offers
    </a>
  </div>
</div>
`;
